common/djangoapps/student/helpers.py

```diff
@@ -182,6 +182,8 @@ def check_verify_status_by_course(user, course_enrollments):
                 now = datetime.now(UTC)
                 if deadline is not None and deadline > now:
                     days_until_deadline = (deadline - now).days
+                elif status == VERIFY_STATUS_NEED_TO_REVERIFY:
```

Comment:
Previously as you can see [here](https://github.com/edx/edx-platform/pull/14694/files#diff-d4bf11344f0cb0d0fb76cd819987a5feL356) it always shows **expiring_soon_window** which was 28, doing this will make sure it show actual number of days left until the expiration of verification